scalanet/src/io/iohk/scalanet/peergroup/UDPPeerGroup.scala

```diff
@@ -83,7 +94,6 @@ class UDPPeerGroup[M](val config: Config, cleanupScheduler: Scheduler = Schedule
               try {
```

Comment:
Should we report the case that any error happened here as `UnexpectedError`? Why are this `try`s needed here?

Reply:
we can. 

But main reason for them is the `finally` block in which realease bytebuffer, that way buffer will be released no matter what error will happen.